File: dart/algorithms/sort/bubble_sort.dart
import 'package:collection/collection.dart';

List<int> bubbleSort(List<int> input) {
  for (int current = 0; current < input.length - 1; current++) {
    for (int next = current + 1; next < input.length; next++) {
      // Swap 2,1 to 1,2
      if (input[current] > input[next]) {
        input.swap(next, current);
      }
    }
  }

  return input;
}

File: kotlin/algorithms/sort/BubbleSort.kt
package algorithms.sort

import java.util.Collections
fun bubbleSort(input: List<Int>): List<Int> {
    for (current in 0 until input.size - 1) {
        for (next in current + 1 until input.size) {
            // Swap 2,1 to 1,2
            if (input[current] > input[next]) {
                Collections.swap(input, current, next)
            }
        }
    }
    return input
}